### PR TITLE
feat(graalvm): support Node.js `http` module

### PR DESCRIPTION
**Pull request by darvld**
Draft

**Coming soon.**

### CODE REVIEW

File: packages/graalvm/build.gradle.kts [hunk 1]
```diff
@@ -127,6 +127,9 @@ dependencies {
   implementation(libs.micronaut.cache.core)
   implementation(libs.micronaut.cache.caffeine)
 
+  // Netty
+  implementation(libs.reactor.netty)
```
Comment (darvld) [inline packages/graalvm/build.gradle.kts:131]: need to swap for a more direct netty dependency (we're not really using reactor anymore, only netty)

File: packages/graalvm/api/graalvm.api [hunk 2]
```diff
@@ -1075,3 +1092,78 @@ public final class elide/runtime/intrinsics/js/err/ValueError$Factory : elide/ru
 	public fun create (Ljava/lang/Throwable;)Lelide/runtime/intrinsics/js/err/ValueError;
 }
 
+public abstract interface class elide/runtime/intrinsics/js/http/HttpAPI {
+	public abstract fun createServer (Ljava/lang/Object;Lorg/graalvm/polyglot/Value;)Lelide/runtime/intrinsics/js/http/Server;
+	public static synthetic fun createServer$default (Lelide/runtime/intrinsics/js/http/HttpAPI;Ljava/lang/Object;Lorg/graalvm/polyglot/Value;ILjava/lang/Object;)Lelide/runtime/intrinsics/js/http/Server;
+}
```
Comment (sgammon) [inline packages/graalvm/api/graalvm.api:1098]: should `internal`ize as many of these classes as possible, lest we provide surface which users may consume

File: packages/graalvm/build.gradle.kts (hunk 1 above)
Comment (sgammon) [reply packages/graalvm/build.gradle.kts:131]: we have a netty bom which can be used with:
```
implementation(platform(libs.netty.bom))
```

then you can simply select whatever netty dep you want to use without declaring a version.

File: packages/graalvm/src/main/kotlin/elide/runtime/gvm/internals/intrinsics/js/http/HttpAPIInstrinsic.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import elide.runtime.gvm.internals.intrinsics.GuestIntrinsic
+import elide.runtime.gvm.internals.intrinsics.Intrinsic
+import elide.runtime.gvm.internals.intrinsics.js.AbstractJsIntrinsic
+import elide.runtime.gvm.internals.intrinsics.js.JsSymbol.JsSymbols.asJsSymbol
+import elide.runtime.intrinsics.js.http.HttpAPI
+import elide.runtime.intrinsics.js.http.Server
+import org.graalvm.polyglot.Value
+
+@Intrinsic internal class HttpAPIInstrinsic : HttpAPI, AbstractJsIntrinsic() {
```
Comment (sgammon) [inline packages/graalvm/src/main/kotlin/elide/runtime/gvm/internals/intrinsics/js/http/HttpAPIInstrinsic.kt:11]: doc ?

File: packages/graalvm/src/main/kotlin/elide/runtime/gvm/internals/intrinsics/js/http/HttpHeadersIntrinsic.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package elide.runtime.gvm.internals.intrinsics.js.http
+
+import elide.runtime.gvm.internals.intrinsics.js.struct.map.JsMutableMultiMap
+import elide.runtime.intrinsics.js.MultiMapLike
+import elide.runtime.intrinsics.js.http.HttpHeaders
+import org.graalvm.polyglot.Value
+import io.netty.handler.codec.http.HttpHeaders as NettyHttpHeaders
+
+@JvmInline
+internal value class HttpHeadersIntrinsic(
```
Comment (sgammon) [inline packages/graalvm/src/main/kotlin/elide/runtime/gvm/internals/intrinsics/js/http/HttpHeadersIntrinsic.kt:10]: i presume the api is evolving, we can add docs near the end

**Review by sgammon — COMMENTED** [review]
a few comments of early feedback. also i re-pinned the `graalvm` api surface to fix that build error, you can do that with:
```
./gradlew :packages:graalvm:apiDump
```

if we want to merge ahead of doc, I'm down, but let's figure out a way to leave a note for ourselves later (todos, issues).

**Comment by darvld** [comment]
Closing this for now as the implementation needs more work

**Pull request closed without merging**